### PR TITLE
Extract CONTRIBUTING.md from README.md

### PR DESCRIPTION
**Pull request by PawelLipski**
README now contains all the things that might in my assessment be ever needed to someone who **builds the project locally** but **does not contribute**.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ln -s ../../scripts/git-hooks/machete-status-branch .git/hooks/machete-status-branch
+ln -s ../../scripts/git-hooks/post-commit .git/hooks/post-commit
+ln -s ../../scripts/run-pre-build-checks .git/hooks/pre-commit
+```
+---
+Consider increasing maximum heap size for the IDE (the default value is 2048 MB). Go to `Help` -> `Change Memory Settings` for this.
+
+## Run/Debug
+
+To run an instance of IDE with Git Machete IntelliJ Plugin execute `:runIde` Gradle task (`Gradle panel` -> `Tasks` -> `intellij` -> `runIde` or `./gradlew runIde`).
```
Comment (mkondratek) [inline CONTRIBUTING.md:28]: It seems that `./gradlew runIde` fails. :confused: Does it happen only to me?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+By default, IntelliJ logs everything with level `INFO` and above into `idea.log` file.
+The exact location depends on IntelliJ version and OS, check `Help` -> `Show Log in Files` to find out.
+
+To enable logging in `DEBUG` level, add selected categories to list in `Help` -> `Diagnostic Tools` -> `Debug Log Settings`.
+The standard practice in Java logging is to use fully-qualified name of each class as category;
+in our case, however, we're only ever using the categories provided above.
+
+## Generate plugin zip
+
+To generate a plugin archive run `:buildPlugin` Gradle task (`Gradle panel` -> `Tasks` -> `intellij` -> `buildPlugin` or `./gradlew buildPlugin`).
```
Comment (mkondratek) [inline CONTRIBUTING.md:51]: same here

**Comment by mkondratek** [comment]
If it comes about contributing I thought (and probably a solution) to **Gradle heap size problem**. 

As it once turned out. We cannot increase memory for Gradle, because of our CI works at its full power now. On the other hand, Gradle tasks are very commonly executed locally. I went to the (a little) trouble of answering the question of how to lift the restriction imposed by Circle Ci.

>  The configuration is a combination of all your gradle.properties files, but if an option is configured in multiple locations, the first one wins:

_Ref:_ https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties

So I think we could provide a suggestion to the guide to copy the existing props file (and prepare `GRADLE_USER_HOME` before) and configure it according to his/her machine possibilities (or even prepare a template `local_gradle.properties`).

We may try it with options suggested here: 
https://medium.com/@wasyl/make-your-gradle-builds-fast-again-ea323ce6a435

**Comment by mkondratek** [comment]
After a quick view with https://scans.gradle.com/ I can say that the change is significant. 

My options:
```

org.gradle.daemon=true
org.gradle.configureondemand=true
org.gradle.jvmargs=-Xmx12g -XX:MaxPermSize=8192m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
```

**Comment by PawelLipski** [comment]
@mkondratek I addressed the Gradle issues on #280

File: CONTRIBUTING.md (hunk 1 above)
Comment (PawelLipski) [reply CONTRIBUTING.md:28]: #280

File: CONTRIBUTING.md (hunk 2 above)
Comment (PawelLipski) [reply CONTRIBUTING.md:51]: #280

File: README.md [hunk 3]
```diff
@@ -1,70 +1,9 @@
...
+This is a port of [git-machete](https://github.com/VirtusLab/git-machete) into an IntelliJ plugin.
 
-To push the rebuilt image, you need write access to [`gitmachete` organization on Docker Hub](https://hub.docker.com/orgs/gitmachete).
+This plugin is available on [JetBrains Marketplace](https://plugins.jetbrains.com/plugin/14221-git-machete).
 
-```
-DOCKER_BUILDKIT=1 BUILDKIT_PROGRESS=plain docker build -t gitmachete/intellij-plugin-ci .
-docker push gitmachete/intellij-plugin-ci
-```
+For details on contributing, see the [guideline](CONTRIBUTING.md).
```
Comment (micpiotrowski) [inline README.md:9]: I think that moving info about building into `CONTRIBUTING.md` may not be a good idea (can be misleading) - someone may want to just build our plugin from source but not contributing.
So maybe split build related information to another file (`BUILDING.md`?) or just leave it as it was - in this file.

Comment (PawelLipski) [reply README.md:9]: Right, good point... I actually have quite a few OSS projects checked out locally:

image

some of which I might want to build myself, without ever contributing.

With that in mind, I revamped README and CONTRIBUTING;  README now contains all the things that might in my assessment be ever needed to someone who builds the project locally but doesn't contribute.

**Pull request merged**